{"events":[{"defined_class":"com.example.accesscontrol.exception.GlobalExceptionHandlerTest$DatabaseIssues","event":"call","id":173882,"lineno":84,"method_id":"handleDatabaseIssue_cannotCreateTx","path":"src/test/java/com/example/accesscontrol/exception/GlobalExceptionHandlerTest.java","receiver":{"class":"com.example.accesscontrol.exception.GlobalExceptionHandlerTest$DatabaseIssues","object_id":1039315422,"value":"com.example.accesscontrol.exception.GlobalExceptionHandlerTest$DatabaseIssues@3df2b1de"},"static":false,"thread_id":1},{"defined_class":"com.example.accesscontrol.exception.GlobalExceptionHandler","event":"call","id":173883,"lineno":72,"method_id":"handleDatabaseConnectionIssue","parameters":[{"class":"org.springframework.transaction.CannotCreateTransactionException","kind":"req","name":"ex","object_id":1125805612,"value":"org.springframework.transaction.CannotCreateTransactionException: db down"}],"path":"src/main/java/com/example/accesscontrol/exception/GlobalExceptionHandler.java","receiver":{"class":"com.example.accesscontrol.exception.GlobalExceptionHandler","object_id":280977365,"value":"com.example.accesscontrol.exception.GlobalExceptionHandler@10bf5fd5"},"static":false,"thread_id":1},{"elapsed":0.0,"event":"return","id":173884,"parent_id":173883,"return_value":{"class":"org.springframework.http.ResponseEntity","object_id":1261595108,"value":"<503 SERVICE_UNAVAILABLE Service Unavailable,{error=Database connection failed. Please try again later., timestamp=2025-08-18T08:59:36.364503200, status=503},[]>"},"thread_id":1},{"elapsed":1.9998E-4,"event":"return","id":173885,"parent_id":173882,"thread_id":1}],"classMap":[{"children":[{"children":[{"children":[{"children":[{"children":[{"labels":[],"location":"src/test/java/com/example/accesscontrol/exception/GlobalExceptionHandlerTest.java:84","name":"handleDatabaseIssue_cannotCreateTx","static":false,"type":"function"}],"location":"src/test/java/com/example/accesscontrol/exception/GlobalExceptionHandlerTest.java","name":"GlobalExceptionHandlerTest$DatabaseIssues","static":false,"type":"class"},{"children":[{"labels":[],"location":"src/main/java/com/example/accesscontrol/exception/GlobalExceptionHandler.java:72","name":"handleDatabaseConnectionIssue","static":false,"type":"function"}],"location":"src/main/java/com/example/accesscontrol/exception/GlobalExceptionHandler.java","name":"GlobalExceptionHandler","static":false,"type":"class"}],"name":"exception","type":"package"}],"name":"accesscontrol","type":"package"}],"name":"example","type":"package"}],"name":"com","type":"package"}],"version":"1.2","metadata":{"name":"Global exception handler $ database issues handle database issue _ cannot create tx","app":"accesscontrol","language":{"name":"java","version":"17.0.15+6-LTS","engine":"OpenJDK 64-Bit Server VM"},"client":{"name":"appmap-java","url":"https://github.com/appland/appmap-java"},"recorder":{"name":"junit","type":"tests"},"recording":{"defined_class":"com.example.accesscontrol.exception.GlobalExceptionHandlerTest$DatabaseIssues","method_id":"handleDatabaseIssue_cannotCreateTx"},"git":{"repository":"https://github.com/AlhaderMajd/accessControl-majd.git","branch":"main","commit":"76e90bbbe9c01451754b3b84b09fec9d285b5774"},"source_location":"src/test/java/com/example/accesscontrol/exception/GlobalExceptionHandlerTest.java:84","frameworks":[{"name":"JUnit","version":"5"}],"test_status":"succeeded"},"eventUpdates":{}}